src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -121,6 +266,100 @@ const DesignColors = () => {
 		}
 	}
 
+	// Pre-defined Colors Section
+
+	/**
+	 * Select a color from predefined palette and sync it.
+	 *
+	 * @param {string} colorStyle - Selected Color Palette slug
+	 */
+	const handleClick = ( colorStyle ) => {
+		if ( selectedColors?.slug === colorStyle ) {
+			return true;
+		}
+
+		clearCustomColors();
+		saveThemeColorPalette( colorStyle );
+		setSelectedColorsLocal( colorPalettes[ colorStyle ] );
+		LocalToState( colorPalettes[ colorStyle ], colorStyle );
+		trackHiiveEvent( 'color-selection', colorStyle );
+	};
+
+	/**
+	 * Build the predefined Color Palette Component
+	 *
+	 * @return {WPComponent} Predefined Color Palette Component
+	 */
+	function buildPredefinedPalette() {
+		return Object.keys( colorPalettes ).map( ( colorStyle, idx ) => {
+			return (
+				<div
+					key={ colorStyle }
+					className={ classNames(
+						'color-palette drawer-palette--button',
+						{
+							'color-palette-selected drawer-palette--button--selected':
+								colorStyle === currentData?.data?.colorStyle,
+						}
+					) }
+					role="button"
+					tabIndex={ idx + 1 }
+					onClick={ () => handleClick( colorStyle ) }
+					onKeyDown={ () => handleClick( colorStyle ) }
+				>
+					<div className="color-palette__colors">
+						<div
+							className="color-palette__colors--tertiary"
+							style={ {
+								backgroundColor: `${ colorPalettes[ colorStyle ]?.[ 'header-background' ] }`,
+							} }
+						/>
+						<div
+							className="color-palette__colors--secondary"
+							style={ {
+								backgroundColor: `${ colorPalettes[ colorStyle ]?.secondary }`,
+							} }
+						/>
+						<div
+							className="color-palette__colors--primary"
+							style={ {
+								backgroundColor: `${ colorPalettes[ colorStyle ]?.primary }`,
+							} }
+						/>
+					</div>
+					<div className="color-palette__name drawer-palette--button__text">
+						{ colorStyle?.charAt( 0 ).toUpperCase() +
```

Comment:
Why use a `?.charAt` when `colorStyle` has already been used for `key`, `onClick` and `onKeyDown`?